react-carousel-component/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  AiOutlineRight,
+  AiOutlineLeft,
+  AiOutlinePayCircle,
+} from 'react-icons/ai';
+import { useEffect, useState } from 'react';
+
+type Prop = {
+  images: {
+    imgLocation: string;
+    id: number;
+  }[];
+};
+//added note to save and commit
+export default function Carousel({ images }: Prop) {
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  useEffect(() => {
+    let interval = setInterval(() => {
+      setActiveIndex((activeIndex + 1) % images.length);
+    }, 3000);
+    return () => clearInterval(interval);
+  }, [activeIndex, images.length]);
+
+  return (
+    <div className="container">
+      <div className="main row">
+        <ArrowButton
+          right={false}
+          onShow={() =>
+            setActiveIndex((activeIndex - 1 + images.length) % images.length)
+          }
+        />
+        <Pokemon images={images} isActive={activeIndex} />
+        <ArrowButton
+          right={true}
+          onShow={() => setActiveIndex((activeIndex + 1) % images.length)}
+        />
+      </div>
+      <Indicators
+        images={images}
+        onSelect={(e) => setActiveIndex(e)}
+        isActive={activeIndex}
+      />
+    </div>
+  );
+}
+
+function ArrowButton({
+  right,
+  onShow,
+}: {
+  right: Boolean;
+  onShow: () => void;
+}) {
+  return (
+    <>
+      {right ? (
+        <AiOutlineRight onClick={onShow} className="right column-10" />
+      ) : (
+        <AiOutlineLeft onClick={onShow} className="left column-10" />
+      )}
+    </>
+  );
+}
+
+type indicatorProps = {
+  images: {
+    imgLocation: string;
+    id: number;
+  }[];
+  onSelect: (id: number) => void;
+  isActive: number;
+};
+
+function Indicators({ images, onSelect, isActive }: indicatorProps) {
+  const buttons: JSX.Element[] = [];
+  images.map((image) => {
+    buttons.push(
+      <AiOutlinePayCircle
+        key={image.id}
+        onClick={() => onSelect(image.id)}
+        className={isActive === image.id ? 'indicator is-on' : 'indicator'}
+      />
+    );
+  });
+
+  return <div className="second row">{buttons}</div>;
+}
+
+type PokemonProps = {
+  images: {
+    imgLocation: string;
+    id: number;
+  }[];
+  isActive: number;
+};
+
+function Pokemon({ images, isActive }: PokemonProps) {
```

Comment:
Interesting choice. If it were me, I would have passed a single image to this component and just displayed it. With this code you are rendering all of them but hiding all except the active one. Why not just pass it `images[activeIndex]` up in the Carousel component?